debugtools/DDR_VM/src/com/ibm/j9ddr/corereaders/macho/MachoDumpReader.java

```diff
@@ -208,10 +209,13 @@ public MachFile64() {}
 		private final List<IMemoryRange> memoryRanges = new LinkedList<>();
 		private final List<IOSStackFrame> stackFrames = new LinkedList<>();
 
-		public OSXThread(long tid, ThreadCommand.ThreadState thread)
+		public OSXThread(long tid, ThreadCommand thread)
 		{
 			threadId = tid;
-			registers = thread.registers;
+			registers = new TreeMap<>();
+			for (ThreadState state : thread.states) {
+				registers.putAll(state.registers);
```

Comment:
Why do we want to merge registers from multiple `ThreadState` objects?

Reply:
They all represent different sets of registers (x86 thread registers, x87 fpu, exception registers, etc). We only handle the x86 thread registers and the exception registers in the core reader, and the registers all have different names so they can be merged. The `IOSThread` interface doesn't have a notion of different sets of registers for a thread so we just merge the different sets.

Reply:
Thanks, @mikezhang1234567890. A comment to that effect would be helpful in the code.